### ISSUE TITLE
Inconsistent MSA representation

### ISSUE THREAD

**Issue by valentas1** [facebookresearch/esm#178]
Hi, 

I am trying to get a representation of an MSA of my protein with the esm_msa1b_t12_100M_UR50S model. However, the representation I am receiving is inconsistent - calling the same function with the same data a few times I am getting slightly different representations. Do you know where this inconsistency could come from?

**Comment by tomsercu** [comment]
Most likely culprit is dropout. Dropout is disabled when the model is in `eval` mode, eg see extract.py#L64  `model.eval()`.
On GPU device there's some non-determinism expected, which can lead to slightly different results. The output should still pass `torch.allclose` check though.

**Comment by tomsercu** [comment]
Feel free to reopen if this didn't answer your question. README was also just updated to default to `model.eval()` to avoid this confusion.

**Comment by valentas1** [comment]
Thank you for the very quick answer, setting model.eval() actually worked.
